lib/autoupdate/tools/helper.go

```diff
@@ -116,12 +116,10 @@ func updateAndReExec(ctx context.Context, updater *Updater, toolsVersion string,
 	// is required if the user passed in the TELEPORT_TOOLS_VERSION
 	// explicitly.
 	err := updater.UpdateWithLock(ctxUpdate, toolsVersion)
-	if err != nil && errors.Is(err, errNoBaseURL) {
+	if errors.Is(err, errNoBaseURL) {
 		// If base URL wasn't defined we have to cancel update and re-execution with warning.
 		slog.WarnContext(ctx, warnMessageOSSBuild)
-		return nil
```

Comment:
This is additional fix, when we disable update we still have to re-execute

Reply:
Why do we need to re-execute when no update occurs?

Reply:
if we have for instance version 16 `oss` in `/usr/local/bin` and previously updated 17 `community` in `~/.tsh/bin`. Next update without env var going to show the message and cancel already updated re-execution

Reply:
So even if we're not downloading a new version, we may need to reexec an already-downloaded version?

Reply:
sorry, didn't notice your comment, yes, this is only one case with double re-execution when v17 from previous example is responsible for remote updated after it is re-executed by v16 already